File: LegacySource/settlement/room/law/execution/ExecutionSingle.cs
using settlement.main;
using settlement.room.main;
using settlement.room.law.execution;
using snake2d.util.datatypes;

namespace settlement.room.law.execution
{
    public class ExecutionSingle : RoomSingleton
    {
        /**
         * 
         */
        private static readonly long serialVersionUID = 1L;

        protected ExecutionSingle(ROOMS m, RoomBlueprint p) : base(m, p)
        {
        }

        public override ROOM_EXECTUTION BlueprintI()
        {
            return SETT.ROOMS().EXECUTION;
        }

        protected override void AddAction(ROOMA ins)
        {
            foreach (COORDINATE c in ins.body())
            {
                if (ins.Is(c))
                {
                    BlueprintI().stations.Init(c.x(), c.y());
                }
            }
            base.AddAction(ins);
        }

        protected override void RemoveAction(ROOMA ins)
        {
            foreach (COORDINATE c in ins.body())
            {
                if (ins.Is(c))
                {
                    BlueprintI().stations.Dispose(c.x(), c.y());
                }
            }
            base.RemoveAction(ins);
        }

        public override void UpdateTileDay(int tx, int ty)
        {
            BlueprintI().stations.Update(tx, ty);
            base.UpdateTileDay(tx, ty);
        }
    }
}

File: settlement/room/law/execution/ExecutionSingle.java
package settlement.room.law.execution;

import settlement.main.SETT;
import settlement.room.main.ROOMA;
import settlement.room.main.ROOMS;
import settlement.room.main.RoomBlueprint;
import settlement.room.main.RoomSingleton;
import snake2d.util.datatypes.COORDINATE;

public class ExecutionSingle extends RoomSingleton{

	/**
	 * 
	 */
	private static final long serialVersionUID = 1L;

	protected ExecutionSingle(ROOMS m, RoomBlueprint p) {
		super(m, p);
	}

	@Override
	public ROOM_EXECTUTION blueprintI() {
		return SETT.ROOMS().EXECUTION;
	}
	
	@Override
	protected void addAction(ROOMA ins) {
		for (COORDINATE c : ins.body()) {
			if (ins.is(c)) {
				blueprintI().stations.init(c.x(), c.y());
			}
		}
		super.addAction(ins);
	}
	
	@Override
	protected void removeAction(ROOMA ins) {
		for (COORDINATE c : ins.body()) {
			if (ins.is(c)) {
				blueprintI().stations.dispose(c.x(), c.y());
			}
		}
		super.removeAction(ins);
	}
	
	@Override
	public void updateTileDay(int tx, int ty) {
		blueprintI().stations.update(tx, ty);
		super.updateTileDay(tx, ty);
	}

}
